src/coreclr/vm/method.hpp

```diff
@@ -1824,38 +1849,54 @@ class MethodDesc
         m_wFlags |= mdfHasNativeCodeSlot;
     }
 
-    inline bool IsAsyncHelperMethod() const
+    // Historically we use "Async2" to mean methods that can be called via CORINFO_CALLCONV_ASYNCCALL
+    // CONSIDER: We could have a better name for the concept, but it is hard to beat shortness of "Async2"
+    inline bool IsAsync2Method() const
     {
         LIMITED_METHOD_DAC_CONTRACT;
         if (!HasAsyncMethodData())
             return false;
         auto asyncKind = GetAddrOfAsyncMethodData()->kind;
-        return asyncKind == AsyncMethodKind::AsyncThunkHelper ||
-            asyncKind == AsyncMethodKind::AsyncImplHelper;
+        return asyncKind == AsyncMethodKind::Async2VariantThunk ||
+            asyncKind == AsyncMethodKind::Async2VariantImpl ||
+            asyncKind == AsyncMethodKind::Async2ExplicitImpl;
```

Comment:
Why did we end up with another kind here?

Reply:
What was `IsAsyncHelperMethod` has become `IsAsync2VariantMethod`, it still checks for two kinds.

I moved `IsAsync2Method` a bit more upfront here, so if someone reads the code sequentially, they will get introduced to `IsAsync2Method` method first, that is simpler thing to understand than `IsAsync2VariantMethod`.

As a result the diff looks like `IsAsync2Method` now checks for one more kind, but that actually did not change.